src/test/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcherTest.java

```diff
@@ -79,4 +92,349 @@ public void testShouldUpdate() throws IOException {
         repository);
     assertFalse(fetcher.shouldUpdate(freshRepository));
   }
+
+  @Test
+  public void testUpdateGithubInstance() throws IOException {
+    GitHub github = mock(GitHub.class);
+    fetcher = spy(new TestGitHubDataFetcher(github));
+    assertEquals(fetcher.github(), github);
+
+    // Create the second GitHub mock bean
+    GitHub github2 = mock(GitHub.class);
+    fetcher = spy(new TestGitHubDataFetcher(github2));
+    assertEquals(fetcher.github(), github2);
+  }
+
+  @Test
+  public void testLoadAndCleanRepository() throws IOException, InterruptedException {
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testfile");
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      localRepositoryFor(project, 1);
+
+      cleanup(project, 0);
+    }
+  }
+
+  @Test
+  public void testCleanAndLoadRepository() throws IOException, InterruptedException {
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testFile");
+
+      cleanup(project, 0);
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      localRepositoryFor(project, 1);
+    }
+  }
+
+  @Test
+  public void testMultiGithubInstanceAndAccessRepo() throws IOException {
+    // Create the first project with first GitHub mock instance.
+    GitHubProject firstProject = new GitHubProject("test", "project1");
+    Path project1Dir = projectDir(firstProject.name());
+    GitHub github = mock(GitHub.class);
+    try (Repository repository =
+        FileRepositoryBuilder.create(project1Dir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, firstProject, "testFile1");
+
+
+      fetcher = spy(new TestGitHubDataFetcher(github));
```

Comment:
Is `spy()` required here? Do you use intercept any calls of `TestGitHubDataFetcher`? If so, why don't you update the test class to do what you need?

Reply:
Updated in https://github.com/SAP/fosstars-rating-core/pull/246#discussion_r448757291